using-mobile-extensions/adobe-target/target-api-reference.md

```diff
@@ -1769,3 +1769,717 @@ ACPCore.collectLaunchInfo(["adb_deeplink" : "com.adobe.targetpreview://app.adobe
 {% endtab %}
 {% endtabs %}
 
+# Public classes
+
+{% tabs %}
+{% tab title="Android" %}
+
+### TargetRequest
+
+Here is a code sample for this class in Android:
+
+```java
+public class TargetRequest extends TargetObject {
+
+    /**
+     * Instantiate a TargetRequest object
+     * @param mboxName String mbox name for this request
+     * @param targetParameters TargetParameters for this request
+     * @param defaultContent String default content for this request
+     * @param contentCallback AdobeCallback<String> which will get called with Target mbox content
+     */
+    public TargetRequest(final String mboxName,
+                         final TargetParameters targetParameters,
+                         final String defaultContent,
+                         final AdobeCallback<String> contentCallback);
+
+     /**
+     * Sets mbox parameters for the request.
+     *
+     * @param mboxParameters Map<String, String> mbox parameters
+     */
+     void setMboxParameters(final Map<String, String> mboxParameters);
+
+     /**
+     * Sets profile parameters for the request.
+     *
+     * @param profileParameters Map<String, String profile parameters
+     */
+    void setProfileParameters(final Map<String, String> profileParameters);
+
+    /**
+     * Sets order parameters for the request.
+     *
+     * @param orderParameters Map<String, Object> order parameters
+     */
+    void setOrderParameters(final Map<String, Object> orderParameters);
+
+    /**
+     * Sets product parameters for the request.
+     *
+     * @param productParameters Map<String, String> product parameters
+     */
+    void setProductParameters(final Map<String, String> productParameters);
+
+    /**
+     * Sets targetParameters for the request.
+     *
+     * @param targetParameters TargetParameters for the request.
+     */
+    void setTargetParameters(final TargetParameters targetParameters);
+}
+```
+
+### TargetPrefetch
+
+Here is a code sample for this class in Android:
+
+```java
+public class TargetPrefetch extends TargetObject {
+
+    /**
+     * Instantiate a TargetPrefetch object
+     * @param mboxName String mbox name for this prefetch request
+     * @param targetParameters TargetParameters for this prefetch request
+     */
+     public TargetPrefetch(final String mboxName, final TargetParameters targetParameters)
+
+     /**
+     * Sets mbox parameters for the request.
+     *
+     * @param mboxParameters Map<String, String> mbox parameters
+     */
+     void setMboxParameters(final Map<String, String> mboxParameters);
```

Comment:
In the Java classes, should we only specify the public methods?

Reply:
i agree, i'll make this change